avocado/plugins/runners/tap.py

```diff
@@ -51,7 +51,7 @@ def _get_tap_result(stdout):
                     result = "skip"
                     break
             elif isinstance(event, TapParser.Test):
-                if event.result in (TestResult.XPASS, TestResult.FAIL):
+                if event.result == TestResult.FAIL:
```

Comment:
I think the current behavior is not terribly wrong.  IIUC, the parser `XPASS` status signals that that an unexpected `ok` was found.  By considering the test as a `fail` (on the Avocado side, the harness here), action can be taken. According to the standard:

"Should a todo test point begin succeeding, the harness may report it in some way that indicates that whatever was supposed to be done has been, and it should be promoted to a normal Test Point."

But, the best solution here, IMO, is to mark those as `WARN`.

Reply:
This makes more sense to me than my solution, I will keep `XPASS` and mark it as `WARN`. Thank you.